remsfal-service/src/main/java/de/remsfal/service/boundary/project/ChatResource.java

```diff
@@ -0,0 +1,371 @@
+package de.remsfal.service.boundary.project;
+
+import de.remsfal.core.api.project.ChatEndpoint;
+import de.remsfal.core.json.project.ChatMessageJson;
+import de.remsfal.core.json.project.ChatSessionJson;
+import de.remsfal.core.model.project.ChatSessionModel;
+import de.remsfal.service.boundary.authentication.RemsfalPrincipal;
+import de.remsfal.service.control.ChatMessageController;
+import de.remsfal.service.control.ChatSessionController;
+import de.remsfal.service.entity.dto.ChatMessageEntity;
+import de.remsfal.service.entity.dto.ChatSessionEntity;
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.core.*;
+import org.jboss.logging.Logger;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+@RequestScoped
+public class ChatResource extends ProjectSubResource implements ChatEndpoint {
+
+    @Inject
+    ChatSessionController chatSessionController;
+
+    @Inject
+    ChatMessageController chatMessageController;
+
+    @Inject
+    RemsfalPrincipal principal;
+
+
+    @Inject
+    Logger logger;
+
+
+    @Override
+    public Response createChatSession() {
+        String userId = principal.getId(); // Get user ID from session
+        if (userId == null || userId.isBlank()) {
+            return Response.status(Response.Status.UNAUTHORIZED)
+                    .entity("User is not authenticated")
+                    .build();
+        }
+
+        String projectId = uri.getPathParameters().getFirst("projectId");
+        String taskId = uri.getPathParameters().getFirst("taskId");
+        String defectId = uri.getPathParameters().getFirst("defectId");
+
+        if ((taskId == null || taskId.isBlank()) && (defectId == null || defectId.isBlank())) {
+            return Response.status(Response.Status.BAD_REQUEST)
```

Comment:
Don't use the happy path to return an error, because you will not trigger a transaction rollback on that way. You should throw a BadRequestException()